src/services/AuthService.ts

```diff
@@ -0,0 +1,69 @@
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
+import { cache } from 'memory-cache';
+
+var JWT_SECRET = 'my-secret-key';
+
+interface User {
+  id: string;
+  email: string;
+  role: string;
+  lastLogin: Date;
+}
+
+export class AuthService {
+  private static instance: AuthService;
+  private users: any = {};
+
+  public static getInstance(): AuthService {
+    if (!AuthService.instance) {
+      AuthService.instance = new AuthService();
+    }
+    return AuthService.instance;
+  }
+
+  async authenticateUser(email: string, password: string) {
+    try {
+      // Call external auth provider
+      const response = await axios.post('http://auth-provider.com/login', {
+        email,
+        password,
+      });
```

Comment:
Can we look at the error responses and give more user friendly errors. A few examples: 
- if the service is down we should retry and then alert the user.  
-If the service returns that we have invalid data, inform the user.
-handle other API errors and log to BE (API key error, rate limiting, etc.)